Code/RDGeneral/Ranking.h

```diff
@@ -72,14 +72,13 @@ void rankVect(const std::vector<T1> &vect, T2 &res) {
   std::sort(indices.begin(), indices.end(), argless<std::vector<T1>>(vect));
 
   int currRank = 0;
-  T1 lastV = vect[indices[0]];
+  unsigned int lastIdx = indices[0];
   BOOST_FOREACH (unsigned int idx, indices) {
```

Comment:
As long as we're in this code, could you please change this to:
```suggestion
  for( auto idx : indices ) {
```